sender-grpc/src/main/java/zipkin2/reporter/otel/grpc/OtelGrpcSender.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2024 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.reporter.otel.grpc;
+
+import static io.grpc.CallOptions.DEFAULT;
+import static io.grpc.MethodDescriptor.generateFullMethodName;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ManagedChannel;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceResponse;
+import io.opentelemetry.proto.trace.v1.TracesData;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import zipkin2.reporter.BytesMessageSender;
+import zipkin2.reporter.ClosedSenderException;
+import zipkin2.reporter.Encoding;
+
+public final class OtelGrpcSender extends BytesMessageSender.Base {
```

Comment:
Just one high level comment, I wonder if we need a gRPC sender. FWIU, all OTel implementations support HTTP (including Zipkin collector here!) and it was made the default transport of OTel at some point post-launch (https://opentelemetry.io/docs/specs/otel/protocol/exporter/#specify-protocol), I would expect future backends to possibly have both or otherwise only support HTTP.

A okhttp-based HTTP sender (AFAICT it's not part of this PR) would feel a lot closer to zipkin's other senders than a gRPC one. It could also potentially be a base for a grpc-less gRPC sender if the need for that came but IMO it's probably not needed here.

Reply:
good point! if we can rule out any protocol for a while, we should. as well we should follow, not lead the norms of spec implementation.

Reply:
I will remove the gRPC stuff entirely then